update community guidelines to exclude building a competitor to us (#5587)
PostHog/posthog.com: contents/docs/contribute/code-of-conduct.md

The following are not allowed within any community space without prior permission 
from the community moderators:

- Pitching of a business, product, or service (outside of the introductions channel). Keep it non spammy!
- Market or competitor research. This is just annoying.
- Encouraging a fork of the main PostHog project or promoting / soliciting people to build a service that competes directly with us. You may be able to do some of this legally, but if successful would harm our project's performance, so it's not in the project's interest to provide a place to promote something like this. If you want to partner, email sales at our company.com, we can work something out.

> Sometimes it's feasible a communtiy member may do something outside of the list above that we've not foreseen, which isn't in the project's interest. We will handle these cases carefully as/if they come up and will edit this guidelines appropriately.

Author: ```suggestion
Sometimes it's feasible a community member may do something outside of the list above that we've not foreseen, which isn't in the project's interest. We will handle these cases carefully as/if they come up and will edit this guidelines appropriately.
```

> Sometimes it's feasible a community member may do something outside of the list above that we've not foreseen, which isn't in the project's interest. We will handle these cases carefully as/if they come up and will edit this guidelines appropriately.

Reviewer: ```suggestion
It's feasible that a community member may do something outside of the list above that we've not foreseen which isn't in the project's best interest. We will handle these cases carefully when they come up and will edit these guidelines accordingly.
```
